src/main/java/com/google/api/generator/engine/ast/ClazzValue.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.generator.engine.ast;
+
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+public abstract class ClazzValue implements ObjectValue {
```

Comment:
We already have a way to achieve the same thing, see the code [here](https://github.com/googleapis/gapic-generator-java/blob/6e02c70bf1b2fdd4ea382427fe8271a00744d915/src/main/java/com/google/api/generator/gapic/composer/common/AbstractServiceStubSettingsClassComposer.java#L331) and the golden test file [here](https://github.com/googleapis/gapic-generator-java/blob/6e02c70bf1b2fdd4ea382427fe8271a00744d915/src/test/java/com/google/api/generator/gapic/composer/grpc/goldens/LoggingServiceV2StubSettings.golden#L463). I think it's better not to have different ways to achieve the same thing, it would create more maintenance burden in the future. If you want, you could extract the common logic to a util method/class, something like `createClazzValueExpr` or `createClazzVariableExpr`, which only takes type as an argument.

Reply:
Ah, thanks for the tip. I couldn't wrap my head around needing a `Value` instead of `Variable`.